src/ol_orchestrate/lib/openedx.py

```diff
@@ -134,3 +143,104 @@ def write_video_json_file(video_data: dict[str, Any], video_data_file: Path):
     with Path(video_data_file).open("w") as video_file:
         video_file.write(json.dumps(video_data))
         video_file.write("\n")
+
+
+def process_course_xml(archive_path: Path) -> dict[str, Any]:
+    """
+    Pull course metadata out of course xml files for edx.org courses
+    (edx.org dumps). Process the course bundle by
+    getting the course_id, finding the metadata file path, calling the
+    function to parse the XML, and then return a dictionary with the metadata elements.
+
+    :param archive_path: The path to the tar archive for the course bundle
+
+    :return: A dictionary with the parsed course metadata.
+    :rtype: Dict[str, Any]
+    """
+    with tarfile.open(archive_path, "r") as tf:
+        tar_info = tf.getmember("course/course.xml")
+        course_xml_file = tf.extract(tar_info)
+        # course xml file is in the root directory
+        course_id, run_tag = parse_course_id(course_xml_file)
+        # use the run_tag to get the course metadata file
+        tar_info = tf.getmember(f"course/course/{run_tag}.xml")
+        course_metadata_file = tf.extract(tar_info)
```

Comment:
So, looking at the documentation again, and the pre-commit errors, the return value of the extract method is `None`. We'll need to change this and the previous `extract` calls to look like:
```suggestion
        course_metadata_file = Path('course_metadata.xml')
        tf.extract(tar_info, path=course_metadata_file)
```